torch/nn/modules/loss.py

```diff
@@ -1119,14 +1119,14 @@ class probabilities only when a single class label per minibatch item is too res
             `Rethinking the Inception Architecture for Computer Vision <https://arxiv.org/abs/1512.00567>`__. Default: :math:`0.0`.
 
     Shape:
-        - Input: :math:`(N, C)` where `C = number of classes`, or
+        - Input: :math:`(N, C)` or :math:`(C)` where `C = number of classes`, or
```

Comment:
The docs here are getting more complex with the added cases, so I think it's important we strive for clarity. Wdyt about following the doc template for [`RNN`](https://pytorch.org/docs/1.9.1/generated/torch.nn.RNN.html)? That is, add a separate section starting with `where:` and list the shapes referenced. This could look something like this (formatting left out, sorry):

```
Input: Shape (C), (N, C), or (N, C, d_1, d_2, ..., d_K) with K >= 1 for K-dimensional loss.
Target: If containing class indices, shape (), (N), or (N, d_1, d_2, ..., d_K) with K >= 1 for K-dimensional loss; each value should be in [0, C-1]. If containing class probabilities, same shape as the input; each value should be in [0, 1].
Output: If reduction is 'none', same shape as the target. Otherwise, scalar.

where:
    C = number of classes
    N = batch size
```

Similar changes for the functional form in `torch/nn/functional.py`.

Reply:
Sure that makes sense. Thanks!